### DISCUSSION TITLE
Audio on localhost?

### DISCUSSION THREAD

**Discussion by Therealjosephchrzempiec** [networked-aframe/networked-aframe#472] [General]
Hello, I got networked-aframe server end with node working on localhost with no problem in windows. Only problem is How do I get audio to work so I can use my microphone? 

Joseph

**Comment by vincentfretin** [comment]
Be sure you use an example that have the settings to use the audio like the basic-audio.html example. You need 
```html
<a-scene
  networked-scene="
      room: basic-audio;
      adapter: easyrtc;
      audio: true;
    "
>
```

**Reply by Therealjosephchrzempiec** [reply]
I have that part. What I mean is without https it will not let the microphone connect.

**Reply by vincentfretin** [reply]
Microphone needs a secure context, localhost, 127.0.0.1 is a secure context or an ip or domain with https. So you need to expose the nodejs server with a self-signed certificate. 
To create the self-signed certificate easily you can use:
```
npx webpack serve --server-type https
```
and ctrl+c
Then to use the self-signed generate certificate edit the `server/easyrtc-server.js` file like this:
```js
const privateKey = fs.readFileSync("node_modules/.cache/webpack-dev-server/server.pem", "utf8");
const certificate = fs.readFileSync("node_modules/.cache/webpack-dev-server/server.pem", "utf8");
```
and uncomment the other relevant lines like the comments say.
and run `npm start`
